extensions/prometheus/prometheus.go

```diff
@@ -0,0 +1,140 @@
+package hogprom
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/exaring/hoglet"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "hoglet"
+)
+
+// WithPrometheusMetrics returns a [hoglet.BreakerMiddleware] that registers prometheus metrics for the circuit breaker.
+//
+// ⚠️ Note: the provided name must be unique across all hoglet instances using the same registerer.
+func WithPrometheusMetrics(name string, reg prometheus.Registerer) hoglet.BreakerMiddleware {
+	return func(next hoglet.ObserverFactory) (hoglet.ObserverFactory, error) {
+		callDurations := prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Namespace: namespace,
+				Subsystem: "circuit",
+				Name:      "call_durations_seconds",
+				Help:      "Call durations in seconds",
+				ConstLabels: prometheus.Labels{
+					"circuit": name,
+				},
+			},
+			[]string{"error"},
```

Comment:
Can we call this label success? It's a success/failure indicator and not a "error"

Reply:
This is fixed now, I'm resolving 😇